src/Target.cpp

```diff
@@ -175,13 +252,38 @@ Target parse_target_string(const std::string &target) {
     t.bits = host.bits;
 
     if (!t.merge_string(target)) {
+        const char *separator = "";
+        std::string architectures;
+        for (auto arch_iter = arch_name_map.begin(); arch_iter != arch_name_map.end(); arch_iter++) {
```

Comment:
Why an old-style for loop here, but a range-based for loop for iterating over arch_name_map below?